docs/localdev.md

```diff
@@ -0,0 +1,89 @@
+# Local Development
+
+## Overview
+
+This document describes howto get a local development enviroment setup, so that a developer can make code changes, fix bugs add new features, test and debug etc for the externalDNS operator \
+We will consider the following scenarios, working with CRC (code ready containers), Kind (kubernetes in Docker) and finally with cluster-bot
+
+Please refer to these links for further info regarding 
+- CRC [link](https://developers.redhat.com/products/cdk/overview)
+- Kind [link](https://kind.sigs.k8s.io/docs/user/quick-start/)
+- Cluster-bot (refer to Slack applications)
+
+
+## CRC (Code Ready Containers)
+
+This section assumes you have downloaded, installed and have a running instance of CRC \
+We will address two modes of working with the external-dns-operator
+- build and push to the internal CRC registry
+- running the external-dns-operator locally
+
+### Build and push to internal CRC registry
+
+1. If you haven't done so create a fork of the openshift/external-dns-operator to your github account
+2. Clone the repo from your github account to your local machine
+3. Change directory `cd <projects-dir>/external-dns-operator`
+4. Ensure your can access your CRC environment
+   ```bash
+   $ export KUBECONFIG=~/.crc/machines/crc/kubeconfig
+   ```
+5. Access to the CRC internal registry
+   ```bash
+   # assume that podman (tool used to create and maintain containers) is our default
+   # assume our default user is kubeadmin
+   $ podman login -u kubeadmin -p $(oc whoami -t) default-route-openshift-image-registry.apps-crc.testing --tls-verify=false
+   ```
+6. We can now follow the standard make recipes
+   ```bash
+   # url format is default-route-openshift-image-registry.apps-crc.testing/<namespace>/<image-name>:tag
+   $ export IMG=default-route-openshift-image-registry.apps-crc.testing/external-dns-operator/external-dns-operator:dev
+   $ CONTAINER_ENGINE=podman make image-build
+   $ TLS_VERFIY=--tls-verify=false make image-push
+   # check that the image was pushed
+   $ oc -n external-dns-operator get is
+   ``` 
+7. Now you can deploy an instance of ExternalDNS
+   ```bash
+   # switch to the external-dns-namespace
+   $ oc project external-dns-operator
```

Comment:
Switching to `external-dns-operator` namespace is not necessary for `make deploy` to work as all the manifests specify the namespace explicitly.
May be the creation of the namespace can be done at the very beginning of the step 6 so that the image stream can be created and all the subsequent `oc` commands will be run in it.

Reply:
I'll ensure that ns is created first

Reply:
The change to the name space is for the oc commands - so that we don't need to type -n external-dns-operator all the time